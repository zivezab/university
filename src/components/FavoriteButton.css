.fav-btn {
    outline: none;
    border: none;
    cursor: pointer;
    color: #6568F4;
}

.fav-btn:hover {
    transition: all 0.3s ease-out;
    background: #fff;
    color: #6568F4;
    transition: 250ms;
}