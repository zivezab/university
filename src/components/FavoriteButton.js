import React, { Component } from "react";
import { VscStarEmpty, VscStarFull } from "react-icons/vsc";
import "./FavoriteButton.css";

class FavoriteButton extends Component {
  render() {
    return this.props.isFavorite ? (
      <VscStarFull
        className="fav-btn float-right"
        size="20"
        onClick={this.props.removeFavorite}
      />
    ) : (
      <VscStarEmpty
        className="fav-btn float-right"
        size="20"
        onClick={this.props.addFavorite}
      />
    );
  }
}

FavoriteButton.defaultProps = {
  addFavorite: () => {},
  removeFavorite: () => {},
  isFavorite: false,
};

export default FavoriteButton;
