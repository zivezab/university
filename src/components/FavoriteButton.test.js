import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';

import FavoriteButton from './FavoriteButton';

test('matches snapshot', () => {
  const wrapper = shallow(<FavoriteButton />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});
