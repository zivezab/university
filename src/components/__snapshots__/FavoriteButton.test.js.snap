// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<VscStarEmpty
  className="fav-btn float-right"
  onClick={[Function]}
  size="20"
/>
`;
